tests/lib/nested.sh

```diff
@@ -797,13 +798,20 @@ EOF
             else 
                 UBUNTU_IMAGE_CHANNEL_ARG=""
             fi
+
+            declare -a UBUNTU_IMAGE_PRESEED_ARGS
```

Comment:
you could use here UBUNTU_IMAGE_PRESEED_ARGS as a string
UBUNTU_IMAGE_PRESEED_ARGS="" and then in the if
UBUNTU_IMAGE_PRESEED_ARGS="--preseed  --preseed-sign-key=\"$NESTED_UBUNTU_IMAGE_PRESEED_KEY\""
So this is more similar as we use the other variables.

Reply:
That didn't work; and I tried something similar before. With the change you suggested I'm getting the following (note the key value is not quoted and has a space in front):

+ UBUNTU_IMAGE_PRESEED_ARGS='--preseed --preseed-sign-key= (test)'
+ /home/gopath/bin/ubuntu-image snap --image-size 10G /home/gopath/src/github.com/snapcore/snapd/tests/lib/assertions/developer1-20-dangerous.model --channel edge --preseed --preseed-sign-key= '(test)' --output-dir /tmp/work-dir/images --snap /tmp/work-dir/assets/pc-kernel_5.4.0-113.127.1_amd64.snap --snap /tmp/work-dir/assets/pc_20-0.4_amd64.snap --snap /home/gopath/src/github.com/snapcore/snapd/tests/nested/manual/core20-preseed/snapd-from-deb.snap --snap /home/gopath/src/github.com/snapcore/snapd/tests/nested/manual/core20-preseed/new-core20.snap

Quoting the entire $UBUNTU_IMAGE_PRESEED_ARGS when invoking u-i breaks args.